00_find_bad_subs.py

```diff
@@ -0,0 +1,70 @@
+"""finds bad subjects and saves them as json.
+
+based on either sensitvity index d'
+or binomial test against chance.
+the latter works bc of a clever task design
+To reproduce, you have to delete the bad subs from config.
+"""
+# %%
+# Imports
+import pandas as pd
+from scipy.stats import binom_test
+
+from config import FPATH_DS, SUBJS
+
+# %%
+# settings
+from utils import calculate_dPrime, get_behavioral_data
+
+# %%
+# Filepaths and settings
+rejectionThreshold = 0.6  # dprime after which we exclude participants
+fpath_ds = FPATH_DS
+overwrite = False
+fpath_der = fpath_ds / "derivatives"
+
+# %%
+
+subex = []
+dPrimeEx = []
+
+subexBino = []
+binoEx = []
+# based on d'
+for sub in SUBJS:
+    fpath_set = fpath_ds / "sourcedata" / "events" / f"EMP{sub:02}_events.csv"
+    behavior_dat = get_behavioral_data(fpath_set)
+    behavior_dat = behavior_dat.assign(
+        right_wrong=[
+            1 if behav == "correctreject" or behav == "hit" else 0
+            for behav in behavior_dat["behavior"]
+        ]
+    )
+
+    dPrime = calculate_dPrime(behavior_dat)  # get dprime
```

Comment:
another (minor) python style comment: in-line comments should be used sparingly -- in this case, `# get dprime` might as well be a comment just above the line